### PR TITLE
Task 9: Rxjs

### PR DESCRIPTION
**Pull request by Dariaky**

### CODE REVIEW

File: src/app/modules/dashboard/components/course-list/course-list.component.ts [hunk 1]
```diff
@@ -13,32 +15,39 @@ export class CourseListComponent implements OnInit, OnChanges {
...
-      })
-  }
+  ngOnInit(): void { }
 
   ngOnChanges(changes: SimpleChanges) {
     if (changes.searchValue.currentValue) {
-      this.courseService.getList(this.coursesStart, null, changes.searchValue.currentValue)
-        .then((courses: ICourse[]) => {
-          this.courses = courses;
+      this._courseService.getList(this.coursesStart, null, changes.searchValue.currentValue)
```
Comment (EvgeniyBlag) [inline src/app/modules/dashboard/components/course-list/course-list.component.ts:31]: I suggest to put showDataLoader before this line. Such way our methods will just return data without unexpected side effects. And on this level we can see where we execute loader methods

File: src/app/modules/dashboard/components/course-list/course-list.component.ts [hunk 2]
```diff
@@ -48,11 +57,14 @@ export class CourseListComponent implements OnInit, OnChanges {
...
-      this.courseService.removeCourse(id)
-        .then(() => {
-          this.courseService.getList()
-            .then((courses: ICourse[]) => {
-              this.courses = courses;
+      this._courseService.removeCourse(id)
+        .subscribe(() => {
+          this._courseService.getList()
+            .subscribe((courses: ICourse[]) => {
+              setTimeout(() => {
```
Comment (EvgeniyBlag) [inline src/app/modules/dashboard/components/course-list/course-list.component.ts]: why we are using setTimeout here?

Comment (Dariaky) [reply src/app/modules/dashboard/components/course-list/course-list.component.ts]: I wanted to mock data loading process in a more demonstrative way :)

File: src/app/modules/dashboard/components/course-list/course-list.component.ts (hunk 1 above)
Comment (Dariaky) [reply src/app/modules/dashboard/components/course-list/course-list.component.ts:31]: will do

**Pull request merged**